src/vconsole/vconsole-setup.c

```diff
@@ -48,58 +48,42 @@
 #define MAX_CONSOLES 63
 
 static bool is_vconsole(int fd) {
-        unsigned char data[1];
+        unsigned char subcode;
 
-        data[0] = TIOCL_GETFGCONSOLE;
-        return ioctl(fd, TIOCLINUX, data) >= 0;
+        subcode = TIOCL_GETFGCONSOLE;
+        return ioctl(fd, TIOCLINUX, &subcode) >= 0;
 }
 
-static bool is_allocated(int fd, int idx) {
-        struct vt_stat vcs = {};
+static bool is_allocated(int idx) {
         char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
 
-        if (fd >= 0) {
-                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
-                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
-                        return false;
-                }
-                idx = vcs.v_active;
-        }
-
         xsprintf(vcname, "/dev/vcs%i", idx);
-        if (access(vcname, F_OK) < 0)
+        return access(vcname, F_OK) == 0;
+}
+
+static bool is_allocated_byfd(int fd) {
+        struct vt_stat vcs = {};
+
+        if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                log_warning_errno(errno, "VT_GETSTATE failed: %m");
                 return false;
-        return true;
+        }
+        return is_allocated(vcs.v_active);
 }
```

Comment:
you just added this function in an earlier commit. I think it would be better if you'd add it in its final form right-away.